packages/design-system/src/components/Modal/Modal.tsx

```diff
@@ -1,88 +1,142 @@
-
-		const onCloseHandler = () => {
-			dialog.hide();
-			if (onClose) {
-				onClose();
-			}
-		};
-
-		const onValidateHandler = () => {
-			if (onValidate) {
-				onValidate();
-			}
-			dialog.hide();
-		};
-
-		return (
-			<>
-				{disclosure && (
-					<S.DialogDisclosure {...dialog} ref={disclosure.ref} {...disclosure.props}>
-						{disclosureProps => React.cloneElement(disclosure, disclosureProps)}
-					</S.DialogDisclosure>
-				)}
-				<S.DialogBackdrop {...dialog}>
-					<ReakitDialog {...dialog} {...props} as={S.Dialog} ref={ref}>
-						<S.DialogHeading>
-							{icon && <Icon name={icon} />}
-							<div>
-								{title && <h1>{title}</h1>}
-								{subtitle && <h2>{subtitle}</h2>}
-							</div>
-						</S.DialogHeading>
-						{children}
-						<S.DialogButtons>
-							<ButtonSecondary onClick={onCloseHandler}>
-								{t('MODAL_CANCEL', 'Cancel')}
-							</ButtonSecondary>
-							<ButtonPrimary onClick={onValidateHandler}>
-								{t('MODAL_VALIDATE', 'Validate')}
-							</ButtonPrimary>
-						</S.DialogButtons>
-					</ReakitDialog>
-				</S.DialogBackdrop>
-			</>
-		);
-	},
-);
-
-export const useModalState = useDialogState;
-
-export const ModalDisclosure = React.forwardRef<React.ReactElement, React.PropsWithChildren<any>>(
-	(props, ref) => <S.DialogDisclosure ref={ref} {...props} />,
-);
-
-export const ModalDialog = React.forwardRef<React.ReactElement, React.PropsWithChildren<any>>(
-	({ children, ...rest }: React.PropsWithChildren<any>, ref) => (
-		<S.DialogBackdrop {...rest}>
-			<S.Dialog {...rest} as={ReakitDialog} ref={ref}>
-				{children}
-			</S.Dialog>
-		</S.DialogBackdrop>
-	),
-);
+function Modal(props: ModalPropsType): ReactElement {
+	const { header, primaryAction, disclosure, onClose, secondaryAction, preventEscaping, children } =
+		props;
+	const hasDisclosure = 'disclosure' in props;
+
+	const dialog = useDialogState({ visible: !hasDisclosure });
+	const ref = useRef(null);
+
+	useEffect(() => {
+		(ref.current as unknown as HTMLElement)?.focus();
+	}, [dialog.visible]);
+
+	const onCloseHandler = hasDisclosure
+		? () => dialog.setVisible(false)
+		: () => onClose && onClose();
+
+	let primaryActionRendered;
+	if (primaryAction) {
+		const dataTest = 'modal.buttons.primary';
+
+		if (!('destructive' in primaryAction) || !primaryAction.destructive) {
+			primaryActionRendered = (
+				<ButtonPrimary {...(primaryAction as ButtonPrimaryPropsType)} data-test={dataTest} />
+			);
+		} else {
+			const { destructive, ...buttonProps } = primaryAction;
+			primaryActionRendered = (
+				<ButtonDestructive {...(buttonProps as ButtonDestructivePropsType)} data-test={dataTest} />
+			);
+		}
+	}
+
+	return (
+		<>
+			{disclosure && (
+				<DialogDisclosure {...dialog}>
+					{disclosureProps => React.cloneElement(disclosure, disclosureProps)}
+				</DialogDisclosure>
+			)}
+			{dialog.visible && (
+				<DialogBackdrop {...dialog} className={styles['modal-backdrop']} data-test="modal.backdrop">
```

Comment:
```suggestion
				<DialogBackdrop {...dialog} className={styles.modal-backdrop} data-test="modal.backdrop">
```
Does that work? That's the way we handle styles in the other files and we'd prefer coding styles to be aligned.

Reply:
Nope because it would evaluate the mathematical expression (_a minus b_)